### ISSUE TITLE
`PyLong` and `PyUnicode` don't match the Python type names

### ISSUE THREAD

**Issue by encukou** [capi-workgroup/problems#14]

**Comment by gvanrossum** [comment]
Could be solved with a (large) number of macros. But is it worth the churn? Ove time, naming never remains fully consistent.

**Comment by encukou** [comment]
IMO, it's not worth the churn at all. But I have seen people trying to name *new* APIs with `Int`/`Str` prefixes. What to do there? IMO, we should be consistent with the existing names, unless we're overhauling *all* of the API.

cc @markshannon

Full disclosure: I maintain a Python 2/3 compat library that [does stuff like alias `PyInt_*` to `PyLong_*`](https://py3c.readthedocs.io/en/latest/defs.html), which is technically abuse of the reserved `Py` namespace. But I don't think I'm the only one doing that.

**Comment by markshannon** [comment]
I agree it isn't worth the churn to change the current API, but it does provide a way to distinguish between old API (`Long`/`Unicode`) and new API (`Int`/`Str`) should we add a new API.

**Comment by serhiy-storchaka** [comment]
What is this new APIs with Int/Str prefixes? "Int" means "related to C `int`", "String" means "related to C `char *`". Having names with "Str" and "String" with very different meaning would confuse.

**Comment by encukou** [comment]
Guideline issue, where I propose to keep the old/consistent names: https://github.com/capi-workgroup/api-evolution/issues/23
